heroic-elasticsearch-utils/src/main/java/com/spotify/heroic/elasticsearch/AbstractElasticsearchMetadataBackend.java

```diff
@@ -106,15 +107,16 @@ public AbstractElasticsearchMetadataBackend(final AsyncFramework async, final St
             final SearchHit[] hits = response.getHits().getHits();
 
             for (final SearchHit hit : hits) {
-                if (limit.isGreater(size)) {
-                    return async.resolved(new LimitedSet<>(limit.limitSet(results), true));
-                }
-
                 if (!results.add(converter.apply(hit))) {
                     duplicates += 1;
+                } else {
+                    size += 1;
                 }
 
-                size += 1;
+                if (limit.isGreater(size)) {
+                    results.remove(converter.apply(hit));
```

Comment:
Not really arguing for performance, but readability yes. It makes it clearer that they are the same instance (or equivalent, in this case).